engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/dto/externaltask/FetchExternalTasksDto.java

```diff
@@ -318,21 +320,24 @@ protected SortMapper(List<SortingDto> sorting, FetchAndLockBuilder builder) {
      * Applies the sorting field mappings to the builder and returns it.
      */
     protected FetchAndLockBuilder getBuilderWithSortConfigs() {
-      sorting.forEach(dto -> {
-        fieldMappingKey(dto).ifPresent(key -> FIELD_MAPPINGS.get(key).accept(builder));
-        orderMappingKey(dto).ifPresent(key -> ORDER_MAPPINGS.get(key).accept(builder));
-      });
-
+      for (SortingDto dto : sorting) {
+        String sortBy = dto.getSortBy();
+        if (FIELD_MAPPINGS.containsKey(sortBy)) {
+          FIELD_MAPPINGS.get(sortBy).accept(builder);
+        } else {
+          throw new InvalidRequestException(Response.Status.BAD_REQUEST, "Cannot set query sortBy parameter to value " + sortBy);
+        }
+
+        String sortOrder = dto.getSortOrder() != null ? dto.getSortOrder().toLowerCase() : null;
```

Comment:
Let's [clarify](https://github.com/camunda/camunda-bpm-platform/issues/4051#issuecomment-2042477967) if the behaviour here should be direct match. If it is, then this line can simply use the dto's value.
```suggestion
        String sortOrder = dto.getSortOrder();
```